Drop StringBuilder:append() overload taking in a `const char*`

cdumez opened the pull request:
#### eb7b0b0b9e5e2cda97a1624334713db993fa4ce0
<pre>
Drop StringBuilder:append() overload taking in a `const char*`
<a href="https://bugs.webkit.org/show_bug.cgi?id=273671">https://bugs.webkit.org/show_bug.cgi?id=273671</a>

Reviewed by Tim Nguyen and Darin Adler.

Drop StringBuilder:append() overload taking in a `const char*`, in favor of the
ones taking in an ASCIILiteral or a std::span.

* Source/JavaScriptCore/API/tests/PingPongStackOverflowTest.cpp:
(PingPongStackOverflowObject_hasInstance):
* Source/JavaScriptCore/bytecode/Instruction.h:
* Source/JavaScriptCore/bytecode/Opcode.cpp:
(JSC::padOpcodeName):
(JSC::OpcodeStats::~OpcodeStats):
* Source/JavaScriptCore/bytecode/Opcode.h:
* Source/JavaScriptCore/bytecompiler/NodesCodegen.cpp:
(JSC::ArrayPatternNode::toString const):
(JSC::RestParameterNode::toString const):
* Source/JavaScriptCore/heap/GCLogging.cpp:
(JSC::GCLogging::levelAsString):
* Source/JavaScriptCore/heap/GCLogging.h:
* Source/JavaScriptCore/heap/HeapSnapshotBuilder.cpp:
(JSC::HeapSnapshotBuilder::json):
* Source/JavaScriptCore/jsc.cpp:
(JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/runtime/ConsoleClient.cpp:
(JSC::appendMessagePrefix):
(JSC::ConsoleClient::printConsoleMessage):
(JSC::ConsoleClient::printConsoleMessageWithArguments):
* Source/JavaScriptCore/runtime/DateConversion.cpp:
(JSC::formatDateTime):
* Source/JavaScriptCore/runtime/ExceptionHelpers.cpp:
(JSC::notAFunctionSourceAppender):
* Source/JavaScriptCore/runtime/FileBasedFuzzerAgent.cpp:
(JSC::FileBasedFuzzerAgent::getPredictionInternal):
* Source/JavaScriptCore/runtime/FileBasedFuzzerAgentBase.cpp:
(JSC::FileBasedFuzzerAgentBase::createLookupKey):
* Source/JavaScriptCore/runtime/FunctionConstructor.cpp:
(JSC::stringifyFunction):
* Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:
(JSC::IntlDateTimeFormat::buildSkeleton):
(JSC::IntlDateTimeFormat::createDateIntervalFormatIfNecessary):
* Source/JavaScriptCore/runtime/IntlDurationFormat.cpp:
(JSC::collectElements):
* Source/JavaScriptCore/runtime/IntlNumberFormat.cpp:
(JSC::IntlNumberFormat::initializeNumberFormat):
* Source/JavaScriptCore/runtime/IntlNumberFormatInlines.h:
(JSC::appendNumberFormatDigitOptionsToSkeleton):
* Source/JavaScriptCore/runtime/Options.cpp:
(JSC::asString):
(JSC::Options::executeDumpOptions):
(JSC::Options::dumpAllOptions):
(JSC::Options::dumpAllOptionsInALine):
(JSC::Options::dumpOption):
(JSC::OptionsHelper::Option::dump const):
* Source/JavaScriptCore/runtime/Options.h:
(JSC::Options::dumpAllOptions):
* Source/JavaScriptCore/runtime/PredictionFileCreatingFuzzerAgent.cpp:
(JSC::PredictionFileCreatingFuzzerAgent::getPredictionInternal):
* Source/JavaScriptCore/runtime/RegExp.cpp:
(JSC::appendLineTerminatorEscape&lt;UChar&gt;):
* Source/JavaScriptCore/runtime/TypeProfiler.cpp:
(JSC::TypeProfiler::typeInformationForExpressionAtOffset):
* Source/JavaScriptCore/runtime/TypeSet.cpp:
(JSC::TypeSet::dumpTypes const):
(JSC::TypeSet::toJSONString const):
(JSC::StructureShape::propertyHash):
(JSC::StructureShape::stringRepresentation):
(JSC::StructureShape::toJSONString const):
* Source/JavaScriptCore/tools/FunctionOverrides.cpp:
(JSC::parseClause):
* Source/WTF/wtf/JSONValues.cpp:
(WTF::JSONImpl::Value::writeJSONImpl const):
* Source/WTF/wtf/text/StringBuilder.h:
(WTF::StringBuilder::append):
* Source/WTF/wtf/text/TextStream.cpp:
(WTF::TextStream::operator&lt;&lt;):
* Source/WebCore/Modules/airplay/WebMediaSessionManager.cpp:
(WebCore::mediaProducerStateString):
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp:
(WebCore::MediaKeySession::updateKeyStatuses):
* Source/WebCore/Modules/encryptedmedia/NavigatorEME.cpp:
(WTF::LogArgument&lt;Vector&lt;T&gt;&gt;::toString):
* Source/WebCore/Modules/fetch/FetchHeaders.cpp:
(WebCore::FetchHeaders::get const):
* Source/WebCore/Modules/geolocation/Geolocation.cpp:
(WebCore::logError):
* Source/WebCore/Modules/indexeddb/IDBKeyData.cpp:
(WebCore::IDBKeyData::loggingString const):
* Source/WebCore/Modules/indexeddb/IDBKeyPath.cpp:
(WebCore::loggingString):
* Source/WebCore/Modules/indexeddb/server/IndexValueStore.cpp:
(WebCore::IDBServer::IndexValueStore::loggingString const):
* Source/WebCore/Modules/indexeddb/shared/IDBDatabaseInfo.cpp:
(WebCore::IDBDatabaseInfo::loggingString const):
* Source/WebCore/Modules/indexeddb/shared/IDBObjectStoreInfo.cpp:
(WebCore::IDBObjectStoreInfo::loggingString const):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCUtils.cpp:
(WebCore::toRTCCodecParameters):
* Source/WebCore/Modules/websockets/WebSocket.cpp:
(WebCore::encodeProtocolString):
* Source/WebCore/Modules/websockets/WebSocketExtensionDispatcher.cpp:
(WebCore::WebSocketExtensionDispatcher::createHeaderValue const):
(WebCore::WebSocketExtensionDispatcher::appendAcceptedExtension):
* Source/WebCore/PAL/pal/avfoundation/OutputContext.mm:
(PAL::OutputContext::deviceName):
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::callInWorld):
* Source/WebCore/contentextensions/CombinedURLFilters.cpp:
(WebCore::ContentExtensions::prefixTreeVertexToString):
(WebCore::ContentExtensions::recursivePrint):
* Source/WebCore/contentextensions/Term.h:
(WebCore::ContentExtensions::Term::toString const):
* Source/WebCore/css/CSSBasicShapes.cpp:
(WebCore::buildEllipseString):
(WebCore::buildInsetString):
* Source/WebCore/css/CSSContainerRule.cpp:
(WebCore::CSSContainerRule::cssText const):
* Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
(WebCore::CSSCounterStyleDescriptors::rangesCSSText const):
(WebCore::CSSCounterStyleDescriptors::symbolsCSSText const):
(WebCore::CSSCounterStyleDescriptors::additiveSymbolsCSSText const):
* Source/WebCore/css/CSSFontFaceSrcValue.cpp:
(WebCore::CSSFontFaceSrcResourceValue::customCSSText const):
* Source/WebCore/css/CSSFontFeatureValuesRule.cpp:
(WebCore::CSSFontFeatureValuesRule::cssText const):
* Source/WebCore/css/CSSFontFeatureValuesRule.h:
* Source/WebCore/css/CSSFontPaletteValuesRule.cpp:
(WebCore::CSSFontPaletteValuesRule::overrideColors const):
(WebCore::CSSFontPaletteValuesRule::cssText const):
* Source/WebCore/css/CSSGradientValue.cpp:
(WebCore::appendGradientStops):
(WebCore::CSSLinearGradientValue::customCSSText const):
(WebCore::CSSPrefixedLinearGradientValue::customCSSText const):
(WebCore::CSSDeprecatedLinearGradientValue::customCSSText const):
(WebCore::CSSRadialGradientValue::customCSSText const):
(WebCore::CSSPrefixedRadialGradientValue::customCSSText const):
(WebCore::CSSDeprecatedRadialGradientValue::customCSSText const):
(WebCore::CSSConicGradientValue::customCSSText const):
* Source/WebCore/css/CSSGroupingRule.cpp:
(WebCore::CSSGroupingRule::appendCSSTextForItemsInternal const):
(WebCore::CSSGroupingRule::cssTextForRules const):
(WebCore::CSSGroupingRule::cssTextForRulesWithReplacementURLs const):
* Source/WebCore/css/CSSImportRule.cpp:
(WebCore::CSSImportRule::cssTextInternal const):
* Source/WebCore/css/CSSLayerBlockRule.cpp:
(WebCore::CSSLayerBlockRule::cssText const):
* Source/WebCore/css/CSSLayerStatementRule.cpp:
(WebCore::CSSLayerStatementRule::cssText const):
* Source/WebCore/css/CSSLineBoxContainValue.cpp:
(WebCore::CSSLineBoxContainValue::customCSSText const):
* Source/WebCore/css/CSSMarkup.cpp:
(WebCore::serializeURL):
* Source/WebCore/css/CSSMediaRule.cpp:
(WebCore::CSSMediaRule::cssText const):
(WebCore::CSSMediaRule::cssTextWithReplacementURLs const):
* Source/WebCore/css/CSSNamespaceRule.cpp:
(WebCore::CSSNamespaceRule::cssText const):
* Source/WebCore/css/CSSPropertyRule.cpp:
(WebCore::CSSPropertyRule::cssText const):
* Source/WebCore/css/CSSScopeRule.cpp:
(WebCore::CSSScopeRule::cssText const):
* Source/WebCore/css/CSSSelectorList.cpp:
(WebCore::CSSSelectorList::buildSelectorsText const):
* Source/WebCore/css/CSSStyleRule.cpp:
(WebCore::CSSStyleRule::cssTextForRules const):
(WebCore::CSSStyleRule::cssTextForRulesWithReplacementURLs const):
(WebCore::CSSStyleRule::cssTextInternal const):
* Source/WebCore/css/CSSStyleSheet.cpp:
(WebCore::CSSStyleSheet::cssTextWithReplacementURLs):
* Source/WebCore/css/CSSSupportsRule.cpp:
(WebCore::CSSSupportsRule::cssText const):
(WebCore::CSSSupportsRule::cssTextWithReplacementURLs const):
* Source/WebCore/css/CSSTimingFunctionValue.cpp:
(WebCore::CSSLinearTimingFunctionValue::customCSSText const):
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::ShorthandSerializer::serializeBorderRadius const):
(WebCore::ShorthandSerializer::serializeGridTemplate const):
* Source/WebCore/css/calc/CSSCalcOperationNode.cpp:
(WebCore::CSSCalcOperationNode::buildCSSText):
* Source/WebCore/css/parser/CSSParserToken.cpp:
(WebCore::CSSParserToken::serialize const):
* Source/WebCore/css/query/GenericMediaQuerySerialization.cpp:
(WebCore::MQ::serialize):
* Source/WebCore/css/query/MediaQueryParser.cpp:
(WebCore::MQ::serialize):
* Source/WebCore/css/typedom/CSSOMVariableReferenceValue.cpp:
(WebCore::CSSOMVariableReferenceValue::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathInvert.cpp:
(WebCore::CSSMathInvert::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathMax.cpp:
(WebCore::CSSMathMax::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathMin.cpp:
(WebCore::CSSMathMin::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathNegate.cpp:
(WebCore::CSSMathNegate::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathProduct.cpp:
(WebCore::CSSMathProduct::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathSum.cpp:
(WebCore::CSSMathSum::serialize const):
* Source/WebCore/css/typedom/transform/CSSMatrixComponent.cpp:
(WebCore::CSSMatrixComponent::serialize const):
* Source/WebCore/css/typedom/transform/CSSPerspective.cpp:
(WebCore::CSSPerspective::serialize const):
* Source/WebCore/css/typedom/transform/CSSRotate.cpp:
(WebCore::CSSRotate::serialize const):
* Source/WebCore/css/typedom/transform/CSSScale.cpp:
(WebCore::CSSScale::serialize const):
* Source/WebCore/css/typedom/transform/CSSSkewX.cpp:
(WebCore::CSSSkewX::serialize const):
* Source/WebCore/css/typedom/transform/CSSSkewY.cpp:
(WebCore::CSSSkewY::serialize const):
* Source/WebCore/css/typedom/transform/CSSTranslate.cpp:
(WebCore::CSSTranslate::serialize const):
* Source/WebCore/dom/Element.cpp:
(WebCore::appendAttributes):
* Source/WebCore/dom/FragmentDirectiveGenerator.cpp:
(WebCore::FragmentDirectiveGenerator::generateFragmentDirective):
* Source/WebCore/dom/SecurityContext.cpp:
(WebCore::SecurityContext::parseSandboxPolicy):
* Source/WebCore/dom/Text.cpp:
(WebCore::appendTextRepresentation):
* Source/WebCore/editing/FontShadow.cpp:
(WebCore::serializationForCSS):
* Source/WebCore/editing/markup.cpp:
(WebCore::StyledMarkupAccumulator::appendStyleNodeOpenTag):
(WebCore::StyledMarkupAccumulator::appendStartTag):
(WebCore::StyledMarkupAccumulator::appendEndTag):
(WebCore::serializePreservingVisualAppearanceInternal):
(WebCore::urlToMarkup):
* Source/WebCore/html/FormController.cpp:
(WebCore::formSignature):
* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::HTMLImageElement::completeURLsInAttributeValue const):
* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::restrictionNames):
* Source/WebCore/html/parser/HTMLSrcsetParser.cpp:
(WebCore::replaceURLsInSrcsetAttribute):
* Source/WebCore/inspector/InspectorFrontendAPIDispatcher.cpp:
(WebCore::expressionForEvaluatingCommand):
* Source/WebCore/inspector/InspectorOverlay.cpp:
(WebCore::InspectorOverlay::buildFlexOverlay):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::InspectorStyleSheet::setRuleStyleText):
* Source/WebCore/loader/ResourceLoadStatistics.cpp:
(WebCore::appendBoolean):
(WebCore::appendHashSet):
(WebCore::appendNavigatorAPIOptionSet):
(WebCore::appendScreenAPIOptionSet):
(WebCore::ResourceLoadStatistics::toString const):
* Source/WebCore/loader/archive/mhtml/MHTMLArchive.cpp:
(WebCore::generateRandomBoundary):
(WebCore::MHTMLArchive::generateMHTMLData):
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp:
(WebCore::CaptionUserPreferencesMediaAF::captionsDefaultFontCSS const):
* Source/WebCore/page/PageSerializer.cpp:
(WebCore::PageSerializer::serializeCSSStyleSheet):
* Source/WebCore/platform/Decimal.cpp:
(WebCore::Decimal::toString const):
* Source/WebCore/platform/graphics/AV1Utilities.cpp:
(WebCore::createAV1CodecParametersString):
* Source/WebCore/platform/graphics/CodecUtilities.cpp:
(WebCore::humanReadableStringFromCodecString):
* Source/WebCore/platform/graphics/ColorInterpolationMethod.cpp:
(WebCore::serializationForCSS):
* Source/WebCore/platform/graphics/HEVCUtilities.cpp:
(WebCore::createDoViCodecParametersString):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::SeekTarget::toString const):
* Source/WebCore/platform/graphics/VP9Utilities.cpp:
(WebCore::createVPCodecParametersString):
* Source/WebCore/platform/graphics/avfoundation/InbandTextTrackPrivateAVF.cpp:
(WebCore::InbandTextTrackPrivateAVF::processCueAttributes):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMInstanceFairPlayStreamingAVFObjC.mm:
(WTF::LogArgument&lt;WebCore::CDMInstanceFairPlayStreamingAVFObjC::Keys&gt;::toString):
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateMock.cpp:
(WebCore::MediaRecorderPrivateMock::videoFrameAvailable):
(WebCore::MediaRecorderPrivateMock::audioSamplesAvailable):
(WebCore::MediaRecorderPrivateMock::generateMockCounterString):
* Source/WebCore/platform/mediastream/RealtimeMediaSourceSettings.cpp:
(WebCore::RealtimeMediaSourceSettings::convertFlagsToString):
* Source/WebCore/platform/text/DateTimeFormat.cpp:
(WebCore::DateTimeFormat::quoteAndAppendLiteral):
* Source/WebCore/platform/text/win/LocaleWin.cpp:
(WebCore::LocaleWin::shortTimeFormat):
* Source/WebCore/rendering/RenderTreeAsText.cpp:
(WebCore::quoteAndEscapeNonPrintables):
(WebCore::nodePosition):
* Source/WebCore/svg/SVGPathStringBuilder.cpp:
(WebCore::SVGPathStringBuilder::moveTo):
(WebCore::SVGPathStringBuilder::lineTo):
(WebCore::SVGPathStringBuilder::lineToHorizontal):
(WebCore::SVGPathStringBuilder::lineToVertical):
(WebCore::SVGPathStringBuilder::curveToCubic):
(WebCore::SVGPathStringBuilder::curveToCubicSmooth):
(WebCore::SVGPathStringBuilder::curveToQuadratic):
(WebCore::SVGPathStringBuilder::curveToQuadraticSmooth):
(WebCore::SVGPathStringBuilder::arcTo):
(WebCore::SVGPathStringBuilder::closePath):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::imageLastDecodingOptions):
(WebCore::Internals::dumpMarkerRects):
(WebCore::Internals::mediaSessionRestrictions const):
(WebCore::Internals::pageMediaState):
(WebCore::appendOffsets):
(WebCore::Internals::scrollSnapOffsets):
(WebCore::Internals::dumpStyleResolvers):
* Source/WebDriver/SessionHost.cpp:
(WebDriver::SessionHost::sendCommandToBackend):
* Source/WebDriver/socket/HTTPServerSocket.cpp:
(WebDriver::HTTPRequestHandler::packHTTPMessage const):
* Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp:
(WGSL::Metal::FunctionDefinitionWriter::emitNecessaryHelpers):
(WGSL::Metal::FunctionDefinitionWriter::visit):
(WGSL::Metal::FunctionDefinitionWriter::generatePackingHelpers):
(WGSL::Metal::FunctionDefinitionWriter::emitPackedVector):
(WGSL::Metal::FunctionDefinitionWriter::serializeVariable):
(WGSL::Metal::FunctionDefinitionWriter::visitArgumentBufferParameter):
(WGSL::Metal::visitArguments):
(WGSL::Metal::emitTextureDimensions):
(WGSL::Metal::emitTextureGather):
(WGSL::Metal::emitTextureGatherCompare):
(WGSL::Metal::emitTextureLoad):
(WGSL::Metal::emitTextureSample):
(WGSL::Metal::emitTextureSampleCompare):
(WGSL::Metal::emitTextureSampleGrad):
(WGSL::Metal::emitTextureSampleLevel):
(WGSL::Metal::emitTextureSampleBaseClampToEdge):
(WGSL::Metal::emitTextureSampleBias):
(WGSL::Metal::emitTextureNumLayers):
(WGSL::Metal::emitTextureNumLevels):
(WGSL::Metal::emitTextureNumSamples):
(WGSL::Metal::emitTextureStore):
(WGSL::Metal::emitStorageBarrier):
(WGSL::Metal::emitTextureBarrier):
(WGSL::Metal::emitWorkgroupBarrier):
(WGSL::Metal::emitWorkgroupUniformLoad):
(WGSL::Metal::atomicFunction):
(WGSL::Metal::emitAtomicLoad):
(WGSL::Metal::emitAtomicStore):
(WGSL::Metal::emitAtomicAdd):
(WGSL::Metal::emitAtomicSub):
(WGSL::Metal::emitAtomicMax):
(WGSL::Metal::emitAtomicMin):
(WGSL::Metal::emitAtomicAnd):
(WGSL::Metal::emitAtomicOr):
(WGSL::Metal::emitAtomicXor):
(WGSL::Metal::emitAtomicExchange):
(WGSL::Metal::emitDistance):
(WGSL::Metal::emitLength):
(WGSL::Metal::emitDegrees):
(WGSL::Metal::emitDynamicOffset):
(WGSL::Metal::emitBitcast):
(WGSL::Metal::emitPack2x16Float):
(WGSL::Metal::emitUnpack2x16Float):
(WGSL::Metal::emitPack4xI8):
(WGSL::Metal::emitPack4xI8Clamp):
(WGSL::Metal::emitUnpack4xI8):
(WGSL::Metal::emitPack4xU8):
(WGSL::Metal::emitPack4xU8Clamp):
(WGSL::Metal::emitQuantizeToF16):
(WGSL::Metal::emitRadians):
(WGSL::Metal::emitUnpack4xU8):
(WGSL::Metal::FunctionDefinitionWriter::serializeBinaryExpression):
(WGSL::Metal::FunctionDefinitionWriter::serializeConstant):
* Source/WebGPU/WGSL/Parser.cpp:
* Source/WebKit/NetworkProcess/Classifier/ITPThirdPartyData.cpp:
(WebKit::ITPThirdPartyData::toString const):
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp:
(WebKit::ResourceLoadStatisticsStore::dumpResourceLoadStatistics):
(WebKit::appendBoolean):
(WebKit::appendNextEntry):
(WebKit::ResourceLoadStatisticsStore::appendSubStatisticList const):
(WebKit::ResourceLoadStatisticsStore::resourceToString const):
* Source/WebKit/NetworkProcess/PrivateClickMeasurement/PrivateClickMeasurementDatabase.cpp:
(WebKit::PCM::Database::privateClickMeasurementToStringForTesting const):
(WebKit::PCM::Database::attributionToStringForTesting const):
* Source/WebKit/NetworkProcess/soup/NetworkProcessSoup.cpp:
(WebKit::buildAcceptLanguages):
* Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp:
(WebKit::CacheStorageManager::representationString):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::NetworkStorageManager::cacheStorageRepresentation):
* Source/WebKit/Shared/Gamepad/GamepadData.cpp:
(WebKit::GamepadData::loggingString const):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewRunJavaScriptWithParams):
* Source/WebKit/UIProcess/API/gtk/DropTargetGtk4.cpp:
(WebKit::DropTarget::accept):
* Source/WebKit/UIProcess/Inspector/socket/RemoteInspectorProtocolHandler.cpp:
(WebKit::RemoteInspectorProtocolHandler::platformStartTask):
* Source/WebKit/UIProcess/Launcher/win/ProcessLauncherWin.cpp:
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/UIProcess/WebBackForwardList.cpp:
(WebKit::WebBackForwardList::loggingString):
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::dispatchDidReachLayoutMilestone):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::dumpHistoryItem):
* Tools/TestWebKitAPI/Tests/WTF/CrossThreadTask.cpp:
(TestWebKitAPI::LifetimeLogger::fullName):
* Tools/TestWebKitAPI/Tests/WTF/StringBuilder.cpp:
(TestWebKitAPI::TEST(StringBuilderTest, Append)):
(TestWebKitAPI::TEST(StringBuilderTest, VariadicAppend)):
(TestWebKitAPI::TEST(StringBuilderTest, ToString)):
(TestWebKitAPI::TEST(StringBuilderTest, ToStringPreserveCapacity)):
(TestWebKitAPI::TEST(StringBuilderTest, Clear)):
(TestWebKitAPI::TEST(StringBuilderTest, Array)):
(TestWebKitAPI::TEST(StringBuilderTest, Resize)):
(TestWebKitAPI::TEST(StringBuilderTest, ToAtomString)):
(TestWebKitAPI::TEST(StringBuilderTest, ToAtomStringOnEmpty)):
* Tools/TestWebKitAPI/Tests/WebCore/TextCodec.cpp:
(TestWebKitAPI::testDecode):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestAutomationSession.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebContext.cpp:
(generateHTMLContent):
* Tools/TestWebKitAPI/cocoa/HTTPServer.mm:
(TestWebKitAPI::HTTPResponse::serialize const):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::dumpResourceURL):
(WTR::dumpErrorDescriptionSuitableForTestResult):
(WTR::dumpFrameScrollPosition):
(WTR::InjectedBundlePage::didReceiveTitleForFrame):
(WTR::InjectedBundlePage::willSendRequestForFrame):
(WTR::InjectedBundlePage::didReceiveResponseForResource):
(WTR::InjectedBundlePage::didFinishLoadForResource):
(WTR::InjectedBundlePage::didFailLoadForResource):
* Tools/WebKitTestRunner/InjectedBundle/atspi/AccessibilityUIElementAtspi.cpp:
(WTR::attributesOfElement):
(WTR::AccessibilityUIElement::helpText const):
(WTR::AccessibilityUIElement::attributedStringForRange):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::findAndDumpWorldLeaks):
(WTR::TestController::decideDestinationWithSuggestedFilename):
(WTR::TestController::decidePolicyForNavigationAction):

Canonical link: <a href="https://commits.webkit.org/278348@main">https://commits.webkit.org/278348@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3dac68541dd1babf9e09e73747a8511eb3795ad4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50259 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29552 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/2556 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53518 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/949 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/52562 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35780 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/587 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/41008 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52358 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27218 "Found 1 new test failure: fast/forms/form-submission-crash-4.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43259 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22107 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/24635 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/512 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8637 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/43588 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/46621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/575 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55103 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/49758 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25355 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/512 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/48419 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/26618 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43443 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/47445 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27480 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/57236 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7267 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/26348 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/11764 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/text/StringBuilder.h:
```diff
@@ -60,10 +60,10 @@ class StringBuilder {
     void append(const String&);
     void append(StringView);
     void append(ASCIILiteral);
+    void append(const char*) = delete; // Pass ASCIILiteral or span instead.
```
You’ll want to do the same kind of thing in `StringConcatenate.h` eventually, for the same reason. After that, you might be able to remove this explicit `delete`.

cdumez replied:
Started work on this: https://github.com/WebKit/WebKit/pull/28161

darinadler commented on Source/JavaScriptCore/runtime/FileBasedFuzzerAgentBase.cpp:
```diff
@@ -40,11 +40,11 @@ String FileBasedFuzzerAgentBase::createLookupKey(const String& sourceFilename, O
     StringBuilder lookupKey;
     lookupKey.append(sourceFilename);
-    lookupKey.append("|");
+    lookupKey.append('|');
     lookupKey.append(opcodeNames[opcodeAliasForLookupKey(opcodeId)]);
-    lookupKey.append("|");
+    lookupKey.append('|');
     lookupKey.append(startLocation);
-    lookupKey.append("|");
+    lookupKey.append('|');
     lookupKey.append(endLocation);
     return lookupKey.toString();
```
This could use `makeString` instead of `StringBuilder`, and it will be more efficient.

darinadler commented on Source/WebCore/css/typedom/CSSOMVariableReferenceValue.cpp:
```diff
@@ -65,7 +65,7 @@ String CSSOMVariableReferenceValue::toString() const
 void CSSOMVariableReferenceValue::serialize(StringBuilder& builder, OptionSet<SerializationArguments> arguments) const
 {
-    builder.append("var(");
+    builder.append("var("_s);
     builder.append(m_variable);
```
Could do these both in a single `append` call, which would be more efficient.

darinadler commented on Source/WebCore/css/typedom/transform/CSSMatrixComponent.cpp:
```diff
@@ -103,53 +103,53 @@ CSSMatrixComponent::CSSMatrixComponent(Ref<DOMMatrixReadOnly>&& matrix, Is2D is2
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->b()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->c()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->d()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->e()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->f()));
-        builder.append(")");
+        builder.append(')');
```
This could be so much more efficient by using only one `append` call and not using `String::number` since append supports numeric formatting without allocating and destroying temporary `StringImpl` objects.

darinadler commented on Source/WebCore/css/typedom/transform/CSSMatrixComponent.cpp:
```diff
@@ -103,53 +103,53 @@ CSSMatrixComponent::CSSMatrixComponent(Ref<DOMMatrixReadOnly>&& matrix, Is2D is2
         builder.append(String::number(m_matrix->m21()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->m22()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->m23()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->m24()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->m31()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->m32()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->m33()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->m34()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->m41()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->m42()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->m43()));
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(String::number(m_matrix->m44()));
-        builder.append(")");
+        builder.append(')');
```
This could be so much more efficient by using only one `append` call and not using `String::number` since append supports numeric formatting without allocating and destroying temporary `StringImpl` objects.

darinadler commented on Source/WebCore/dom/FragmentDirectiveGenerator.cpp:
```diff
@@ -115,7 +115,7 @@ void FragmentDirectiveGenerator::generateFragmentDirective(const SimpleRange& te
         StringBuilder textDirectiveBuilder;
         textDirectiveBuilder.append(textDirectivePrefix);
         textDirectiveBuilder.append(startText);
-        textDirectiveBuilder.append(",");
+        textDirectiveBuilder.append(',');
         textDirectiveBuilder.append(endText);
```
This could be done more efficiently with `makeString` instead of `StringBuilder`.

darinadler commented on Source/WebCore/dom/FragmentDirectiveGenerator.cpp:
```diff
@@ -132,9 +132,9 @@ void FragmentDirectiveGenerator::generateFragmentDirective(const SimpleRange& te
         StringBuilder textDirectiveBuilder;
         textDirectiveBuilder.append(textDirectivePrefix);
         textDirectiveBuilder.append(percentEncodeFragmentDirectiveSpecialCharacters(prefix));
-        textDirectiveBuilder.append("-,");
+        textDirectiveBuilder.append("-,"_s);
         textDirectiveBuilder.append(percentEncodeFragmentDirectiveSpecialCharacters(textFromRange));
-        textDirectiveBuilder.append(",-");
+        textDirectiveBuilder.append(",-"_s);
         textDirectiveBuilder.append(percentEncodeFragmentDirectiveSpecialCharacters(suffix));
```
This could be done more efficiently with `makeString` instead of `StringBuilder`.

darinadler commented on Source/WebCore/inspector/InspectorOverlay.cpp:
```diff
@@ -2047,14 +2047,14 @@ std::optional<InspectorOverlay::Highlight::FlexHighlightOverlay> InspectorOverla
 
                 if (auto index = renderChildrenInDOMOrder.find(renderChild); index != notFound) {
-                    orderNumbers.append("Item #");
+                    orderNumbers.append("Item #"_s);
                     orderNumbers.append(index + 1);
```
This could be done more efficiently with a single `append` call.

darinadler commented on Source/WebCore/inspector/InspectorOverlay.cpp:
```diff
@@ -2047,14 +2047,14 @@ std::optional<InspectorOverlay::Highlight::FlexHighlightOverlay> InspectorOverla
 
                 if (auto order = renderChild->style().order(); order || hasCustomOrder) {
                     if (!orderNumbers.isEmpty())
                         orderNumbers.append('\n');
-                    orderNumbers.append("order: ");
+                    orderNumbers.append("order: "_s);
                     orderNumbers.append(order);
```
This could be done more efficiently with a single `append` call.

darinadler commented on Source/WebCore/loader/ResourceLoadStatistics.cpp:
```diff
@@ -477,7 +477,7 @@ String ResourceLoadStatistics::toString() const
     builder.append('\n');
     appendBoolean(builder, "isVeryPrevalentResource"_s, isVeryPrevalentResource);
     builder.append('\n');
-    builder.append("    dataRecordsRemoved: ", dataRecordsRemoved);
+    builder.append("    dataRecordsRemoved: "_s, dataRecordsRemoved);
     builder.append('\n');
```
This could be done more efficiently with a single `append` call.

darinadler commented on Source/WebCore/loader/archive/mhtml/MHTMLArchive.cpp:
```diff
@@ -144,25 +144,25 @@ Ref<FragmentedSharedBuffer> MHTMLArchive::generateMHTMLData(Page* page)
-    stringBuilder.append("\r\nDate: ");
+    stringBuilder.append("\r\nDate: "_s);
     stringBuilder.append(dateString);
-    stringBuilder.append("\r\nMIME-Version: 1.0\r\n");
-    stringBuilder.append("Content-Type: multipart/related;\r\n");
+    stringBuilder.append("\r\nMIME-Version: 1.0\r\n"_s);
+    stringBuilder.append("Content-Type: multipart/related;\r\n"_s);
```
This could be done more efficiently with a single `append` call.

darinadler commented on Source/WebCore/loader/archive/mhtml/MHTMLArchive.cpp:
```diff
@@ -144,25 +144,25 @@ Ref<FragmentedSharedBuffer> MHTMLArchive::generateMHTMLData(Page* page)
+    stringBuilder.append("Content-Type: multipart/related;\r\n"_s);
     if (localMainFrame) {
-        stringBuilder.append("\ttype=\"");
+        stringBuilder.append("\ttype=\""_s);
         stringBuilder.append(localMainFrame->document()->suggestedMIMEType());
```
This could be done more efficiently with a single `append` call.

darinadler commented on Source/WebCore/loader/archive/mhtml/MHTMLArchive.cpp:
```diff
@@ -144,25 +144,25 @@ Ref<FragmentedSharedBuffer> MHTMLArchive::generateMHTMLData(Page* page)
-    stringBuilder.append("\";\r\n");
-    stringBuilder.append("\tboundary=\"");
+    stringBuilder.append("\";\r\n"_s);
+    stringBuilder.append("\tboundary=\""_s);
     stringBuilder.append(boundary);
-    stringBuilder.append("\"\r\n\r\n");
+    stringBuilder.append("\"\r\n\r\n"_s);
```
This could be done more efficiently with a single `append` call.

darinadler commented on Source/WebCore/loader/archive/mhtml/MHTMLArchive.cpp:
```diff
@@ -144,25 +144,25 @@ Ref<FragmentedSharedBuffer> MHTMLArchive::generateMHTMLData(Page* page)
-    stringBuilder.append("\"\r\n\r\n");
+    stringBuilder.append("\"\r\n\r\n"_s);
 
     // We use utf8() below instead of ascii() as ascii() replaces CRLFs with ?? (we still only have put ASCII characters in it).
```
This comment is kind of obsolete since we don’t really have a `String::ascii()` function that should be used with anything, so we don’t have to explain why we don’t use it.

(We can use a much more efficient idiom when the string is almost certainly 8-bit, but I guess we don’t have it packaged up for use.)

darinadler commented on Source/WebCore/platform/graphics/MediaPlayer.cpp:
```diff
@@ -2099,11 +2099,11 @@ String MediaPlayer::lastErrorMessage() const
 {
     StringBuilder builder;
-    builder.append("[");
+    builder.append('[');
     builder.append(WTF::LogArgument<MediaTime>::toString(time));
     builder.append(WTF::LogArgument<MediaTime>::toString(negativeThreshold));
     builder.append(WTF::LogArgument<MediaTime>::toString(positiveThreshold));
-    builder.append("]");
+    builder.append(']');
     return builder.toString();
```
This could use `makeString` instead of `StringBuilder`, which would be more efficient.

darinadler commented on Source/WebCore/platform/graphics/VP9Utilities.cpp:
```diff
@@ -296,7 +296,7 @@ String createVPCodecParametersString(const VPCodecConfigurationRecord& configura
         return resultBuilder.toString();
 
-    resultBuilder.append(".0");
+    resultBuilder.append(".0"_s);
     resultBuilder.append(numberToStringUnsigned<String>(configuration.profile));
```
Should be able to use just one `append` function call here with multiple arguments and should be able to just pass the integer, since `StringBuilder`’s `append` uses `StringConcatenate.h` to support types like integers.

darinadler commented on Source/WebCore/platform/graphics/VP9Utilities.cpp:
```diff
@@ -319,7 +319,7 @@ String createVPCodecParametersString(const VPCodecConfigurationRecord& configura
         return resultBuilder.toString();
 
-    resultBuilder.append(".0");
+    resultBuilder.append(".0"_s);
     resultBuilder.append(numberToStringUnsigned<String>(configuration.chromaSubsampling));
```
Should be able to use just one `append` function call here with multiple arguments and should be able to just pass the integer, since `StringBuilder`’s `append` uses `StringConcatenate.h` to support types like integers.

darinadler commented on Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp:
```diff
@@ -719,8 +719,8 @@ void MediaKeySession::updateKeyStatuses(CDMInstanceSession::KeyStatusVector&& in
-            statusString.append(", ");
-        statusString.append(makeString(convertEnumerationToString(statusCount.key), ": ", statusCount.value));
+            statusString.append(", "_s);
+        statusString.append(makeString(convertEnumerationToString(statusCount.key), ": "_s, statusCount.value));
```
I don’t think we need to call `makeString` here. Should work, more efficiently, without it?

darinadler commented on Source/WebCore/Modules/geolocation/Geolocation.cpp:
```diff
@@ -335,17 +335,17 @@ int Geolocation::watchPosition(Ref<PositionCallback>&& successCallback, RefPtr<P
-    message.append("[blocked] Access to geolocation was blocked over");
+    message.append("[blocked] Access to geolocation was blocked over"_s);
     
     if (!isSecure)
-        message.append(" insecure connection to ");
+        message.append(" insecure connection to "_s);
     else if (isMixedContent)
-        message.append(" secure connection with mixed content to ");
+        message.append(" secure connection with mixed content to "_s);
     else
         return;
     
     message.append(target);
-    message.append(".\n");
+    message.append(".\n"_s);
```
Could use `makeString` here instead of `StringBuilder` and it would be more efficient.

darinadler commented on Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
```diff
@@ -523,9 +523,9 @@ String CSSCounterStyleDescriptors::additiveSymbolsCSSText() const
-            builder.append(", ");
+            builder.append(", "_s);
         builder.append(m_additiveSymbols[i].second);
-        builder.append(" ");
+        builder.append(" "_s);
         builder.append(m_additiveSymbols[i].first.cssText());
```
Could do this more efficiently with a single `append` call.

darinadler commented on Source/WebCore/css/CSSStyleRule.cpp:
```diff
@@ -181,37 +181,37 @@ String CSSStyleRule::cssTextWithReplacementURLs(const HashMap<String, String>& r
 
     if (rules.isEmpty()) {
         builder.append(' ');
         builder.append(declarations);
-        builder.append(" }");
+        builder.append(" }"_s);
```
Could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/css/CSSStyleRule.cpp:
```diff
@@ -181,37 +181,37 @@ String CSSStyleRule::cssTextWithReplacementURLs(const HashMap<String, String>& r
 {
     StringBuilder builder;
     builder.append(selectorText());
-    builder.append(" {");
+    builder.append(" {"_s);
```
Could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/css/CSSStyleRule.cpp:
```diff
@@ -181,37 +181,37 @@ String CSSStyleRule::cssTextWithReplacementURLs(const HashMap<String, String>& r
 
     if (declarations.isEmpty()) {
         builder.append(rules);
-        builder.append("\n}");
+        builder.append("\n}"_s);
```
Could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/css/CSSStyleRule.cpp:
```diff
@@ -181,37 +181,37 @@ String CSSStyleRule::cssTextWithReplacementURLs(const HashMap<String, String>& r
 
-    builder.append("\n  ");
+    builder.append("\n  "_s);
     builder.append(declarations);
     builder.append(rules);
-    builder.append("\n}");
+    builder.append("\n}"_s);
```
Could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/css/query/GenericMediaQuerySerialization.cpp:
```diff
@@ -108,7 +108,7 @@ void serialize(StringBuilder& builder, const Feature& feature)
         serializeIdentifier(feature.name, builder);
 
-        builder.append(": ");
+        builder.append(": "_s);
         builder.append(feature.rightComparison->value->cssText());
```
Could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/editing/FontShadow.cpp:
```diff
@@ -41,11 +41,11 @@ String serializationForCSS(const FontShadow& shadow)
     StringBuilder builder;
-    builder.append(shadow.offset.width(), "px ");
-    builder.append(shadow.offset.height(), "px ");
+    builder.append(shadow.offset.width(), "px "_s);
+    builder.append(shadow.offset.height(), "px "_s);
     builder.append(serializationForCSS(shadow.color));
```
Could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/platform/graphics/CodecUtilities.cpp:
```diff
@@ -44,49 +44,49 @@ String humanReadableStringFromCodecString(const String& codecString)
-        builder.append(")");
+        builder.append(')');
 
         return builder.toString();
```
Could be more efficient by using fewer calls to `append` or using `makeString`.

darinadler commented on Source/WebCore/platform/graphics/CodecUtilities.cpp:
```diff
@@ -44,49 +44,49 @@ String humanReadableStringFromCodecString(const String& codecString)
         switch (parameters->profileIDC) {
         case 66:
             builder.append(WEB_UI_STRING_KEY("Baseline Profile", "Baseline Profile (AVC Codec Profile)", "Codec Strings"));
-            builder.append(", ");
+            builder.append(", "_s);
```
Could be more efficient by using fewer calls to `append`.

darinadler commented on Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp:
```diff
@@ -571,29 +571,29 @@ Ref<TextureMapperShaderProgram> TextureMapperShaderProgram::create(TextureMapper
     // Append the appropriate input/output variable definitions.
-    vertexShaderBuilder.append(vertexTemplateLT320Vars);
+    vertexShaderBuilder.append(span(vertexTemplateLT320Vars));
 
     // Append the common code.
-    vertexShaderBuilder.append(vertexTemplateCommon);
+    vertexShaderBuilder.append(span(vertexTemplateCommon));
```
This could use `makeString` instead of `StringBuilder` and be more efficient.

darinadler commented on Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp:
```diff
@@ -571,29 +571,29 @@ Ref<TextureMapperShaderProgram> TextureMapperShaderProgram::create(TextureMapper
-    vertexShaderBuilder.append(vertexTemplateCommon);
+    vertexShaderBuilder.append(span(vertexTemplateCommon));
 
     StringBuilder fragmentShaderBuilder;
```
This could use `makeString` instead of `StringBuilder` and be more efficient.

darinadler commented on Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp:
```diff
@@ -571,29 +571,29 @@ Ref<TextureMapperShaderProgram> TextureMapperShaderProgram::create(TextureMapper
     StringBuilder fragmentShaderBuilder;
 
     // Append the options.
     fragmentShaderBuilder.append(optionsApplierBuilder.toString());
```
Calling `toString` here makes things less efficient, and should not be needed.

darinadler commented on Source/WebCore/platform/gtk/SelectionData.cpp:
```diff
@@ -109,12 +109,12 @@ void SelectionData::setURL(const URL& url, const String& label)
     StringBuilder markup;
-    markup.append("<a href=\"");
+    markup.append("<a href=\""_s);
     markup.append(url.string());
-    markup.append("\">");
+    markup.append("\">"_s);
     GUniquePtr<gchar> escaped(g_markup_escape_text(actualLabel.utf8().data(), -1));
     markup.append(String::fromUTF8(escaped.get()));
-    markup.append("</a>");
+    markup.append("</a>"_s);
     setMarkup(markup.toString());
```
This could use `makeString` instead of `StringBuilder` and be more efficient.

darinadler commented on Source/WebCore/platform/mediarecorder/MediaRecorderPrivateMock.cpp:
```diff
@@ -71,7 +71,7 @@ void MediaRecorderPrivateMock::resumeRecording(CompletionHandler<void()>&& compl
 {
     Locker locker { m_bufferLock };
-    m_buffer.append("Video Track ID: ");
+    m_buffer.append("Video Track ID: "_s);
     m_buffer.append(m_videoTrackID);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/platform/mediarecorder/MediaRecorderPrivateMock.cpp:
```diff
@@ -82,14 +82,14 @@ void MediaRecorderPrivateMock::audioSamplesAvailable(const MediaTime&, const Pla
     DisableMallocRestrictionsForCurrentThreadScope disableMallocRestrictions;
     Locker locker { m_bufferLock };
-    m_buffer.append("Audio Track ID: ");
+    m_buffer.append("Audio Track ID: "_s);
     m_buffer.append(m_audioTrackID);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/platform/network/curl/NetworkStorageSessionCurl.cpp:
```diff
@@ -74,10 +74,10 @@ static std::pair<String, bool> cookiesForSession(const NetworkStorageSession& se
+                cookies.append("; "_s);
             if (!cookie.name.isEmpty()) {
                 cookies.append(cookie.name);
-                cookies.append("=");
+                cookies.append('=');
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/platform/text/win/LocaleWin.cpp:
```diff
@@ -323,7 +323,7 @@ String LocaleWin::shortTimeFormat()
         format = getLocaleInfoString(LOCALE_STIMEFORMAT);
         StringBuilder builder;
         builder.append(getLocaleInfoString(LOCALE_STIME));
-        builder.append("ss");
+        builder.append("ss"_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/testing/Internals.cpp:
```diff
@@ -1183,16 +1183,16 @@ AtomString Internals::imageLastDecodingOptions(HTMLImageElement& element)
-        builder.append(", sizeForDrawing : { ");
+        builder.append(", sizeForDrawing : { "_s);
         builder.append(sizeForDrawing->width());
-        builder.append(", ");
+        builder.append(", "_s);
         builder.append(sizeForDrawing->height());
-        builder.append(" }");
+        builder.append(" }"_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/testing/Internals.cpp:
```diff
@@ -1183,16 +1183,16 @@ AtomString Internals::imageLastDecodingOptions(HTMLImageElement& element)
     StringBuilder builder;
-    builder.append("{ decodingMode : ");
-    builder.append(options.decodingMode() == DecodingMode::Asynchronous ? "Asynchronous" : "Synchronous");
+    builder.append("{ decodingMode : "_s);
+    builder.append(options.decodingMode() == DecodingMode::Asynchronous ? "Asynchronous"_s : "Synchronous"_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebCore/testing/Internals.cpp:
```diff
@@ -7220,16 +7220,16 @@ String Internals::dumpStyleResolvers()
-        result.append("(identifier=", identifier, ") ");
-        result.append("(author rule count=", resolver.ruleSets().authorStyle().ruleCount(), ")");
-        result.append(")\n");
+        result.append('(', name, ' ');
+        result.append("(identifier="_s, identifier, ") "_s);
+        result.append("(author rule count="_s, resolver.ruleSets().authorStyle().ruleCount(), ')');
+        result.append(")\n"_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp:
```diff
@@ -187,253 +187,253 @@ void FunctionDefinitionWriter::emitNecessaryHelpers()
-            m_stringBuilder.append(m_indent, "float4x3 ColorSpaceConversionMatrix;\n");
-            m_stringBuilder.append(m_indent, "uint get_width(uint lod = 0) const { return FirstPlane.get_width(lod); }\n");
-            m_stringBuilder.append(m_indent, "uint get_height(uint lod = 0) const { return FirstPlane.get_height(lod); }\n");
+            m_stringBuilder.append(m_indent, "texture2d<float> FirstPlane;\n"_s);
+            m_stringBuilder.append(m_indent, "texture2d<float> SecondPlane;\n"_s);
+            m_stringBuilder.append(m_indent, "float3x2 UVRemapMatrix;\n"_s);
+            m_stringBuilder.append(m_indent, "float4x3 ColorSpaceConversionMatrix;\n"_s);
+            m_stringBuilder.append(m_indent, "uint get_width(uint lod = 0) const { return FirstPlane.get_width(lod); }\n"_s);
+            m_stringBuilder.append(m_indent, "uint get_height(uint lod = 0) const { return FirstPlane.get_height(lod); }\n"_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp:
```diff
@@ -187,253 +187,253 @@ void FunctionDefinitionWriter::emitNecessaryHelpers()
-        m_stringBuilder.append(m_indent, "template<typename T, size_t N>\n");
-        m_stringBuilder.append(m_indent, "array<typename T::PackedType, N> __pack(array<T, N> unpacked)\n");
-        m_stringBuilder.append(m_indent, "{\n");
+        m_stringBuilder.append(m_indent, "template<typename T, size_t N>\n"_s);
+        m_stringBuilder.append(m_indent, "array<typename T::PackedType, N> __pack(array<T, N> unpacked)\n"_s);
+        m_stringBuilder.append(m_indent, "{\n"_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp:
```diff
@@ -187,253 +187,253 @@ void FunctionDefinitionWriter::emitNecessaryHelpers()
             IndentationScope scope(m_indent);
-            m_stringBuilder.append(m_indent, "array<typename T::PackedType, N> packed;\n");
-            m_stringBuilder.append(m_indent, "for (size_t i = 0; i < N; ++i)\n");
+            m_stringBuilder.append(m_indent, "array<typename T::PackedType, N> packed;\n"_s);
+            m_stringBuilder.append(m_indent, "for (size_t i = 0; i < N; ++i)\n"_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp:
```diff
@@ -187,253 +187,253 @@ void FunctionDefinitionWriter::emitNecessaryHelpers()
-        m_stringBuilder.append(m_indent, "template<typename T, size_t N>\n");
-        m_stringBuilder.append(m_indent, "array<typename T::UnpackedType, N> __unpack(array<T, N> packed)\n");
-        m_stringBuilder.append(m_indent, "{\n");
+        m_stringBuilder.append(m_indent, "template<typename T, size_t N>\n"_s);
+        m_stringBuilder.append(m_indent, "array<typename T::UnpackedType, N> __unpack(array<T, N> packed)\n"_s);
+        m_stringBuilder.append(m_indent, "{\n"_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp:
```diff
@@ -187,253 +187,253 @@ void FunctionDefinitionWriter::emitNecessaryHelpers()
             IndentationScope scope(m_indent);
-            m_stringBuilder.append(m_indent, "array<typename T::UnpackedType, N> unpacked;\n");
-            m_stringBuilder.append(m_indent, "for (size_t i = 0; i < N; ++i)\n");
+            m_stringBuilder.append(m_indent, "array<typename T::UnpackedType, N> unpacked;\n"_s);
+            m_stringBuilder.append(m_indent, "for (size_t i = 0; i < N; ++i)\n"_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp:
```diff
@@ -187,253 +187,253 @@ void FunctionDefinitionWriter::emitNecessaryHelpers()
-        m_stringBuilder.append(m_indent, "template<typename T>\n");
-        m_stringBuilder.append(m_indent, "T __workgroup_uniform_load(threadgroup T* const ptr)\n");
-        m_stringBuilder.append(m_indent, "{\n");
+        m_stringBuilder.append(m_indent, "template<typename T>\n"_s);
+        m_stringBuilder.append(m_indent, "T __workgroup_uniform_load(threadgroup T* const ptr)\n"_s);
+        m_stringBuilder.append(m_indent, "{\n"_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebGPU/WGSL/Parser.cpp:
```diff
@@ -101,9 +101,9 @@ struct TemplateTypes<TT> {
     if (!name##Expected) { \
         StringBuilder builder; \
-        builder.append("Expected a "); \
+        builder.append("Expected a "_s); \
         builder.append(toString(TokenType::type)); \
-        builder.append(", but got a "); \
+        builder.append(", but got a "_s); \
         builder.append(toString(name##Expected.error())); \
         FAIL(builder.toString()); \
```
This could use `makeString` instead of `StringBuilder` and be more efficient.

darinadler commented on Source/WebGPU/WGSL/Parser.cpp:
```diff
@@ -114,9 +114,9 @@ struct TemplateTypes<TT> {
         if (!expectedToken) { \
             StringBuilder builder; \
-            builder.append("Expected a "); \
+            builder.append("Expected a "_s); \
             builder.append(toString(TokenType::type)); \
-            builder.append(", but got a "); \
+            builder.append(", but got a "_s); \
             builder.append(toString(expectedToken.error())); \
             FAIL(builder.toString()); \
```
This could use `makeString` instead of `StringBuilder` and be more efficient.

darinadler commented on Source/WebGPU/WGSL/Parser.cpp:
```diff
@@ -126,9 +126,9 @@ struct TemplateTypes<TT> {
     if (!name##Expected) { \
         StringBuilder builder; \
-        builder.append("Expected one of ["); \
+        builder.append("Expected one of ["_s); \
         TemplateTypes<__VA_ARGS__>::appendNameTo(builder); \
-        builder.append("], but got a "); \
+        builder.append("], but got a "_s); \
         builder.append(toString(name##Expected.error())); \
         FAIL(builder.toString()); \
```
This could use `makeString` instead of `StringBuilder` and be more efficient.

darinadler commented on Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
```diff
@@ -2098,15 +2098,15 @@ void didCompleteWithError(const WebCore::ResourceError& error, const WebCore::Ne
+    debugString.append("Submitting potential install attribution for AdamId: "_s);
     debugString.append(makeString(*pcm.adamID()));
-    debugString.append(", adNetworkRegistrableDomain: ");
+    debugString.append(", adNetworkRegistrableDomain: "_s);
     debugString.append(pcm.destinationSite().registrableDomain.string());
-    debugString.append(", impressionId: ");
+    debugString.append(", impressionId: "_s);
     debugString.append(pcm.ephemeralSourceNonce()->nonce);
-    debugString.append(", sourceWebRegistrableDomain: ");
+    debugString.append(", sourceWebRegistrableDomain: "_s);
     debugString.append(pcm.sourceSite().registrableDomain.string());
-    debugString.append(", version: 3");
+    debugString.append(", version: 3"_s);
```
This could use `makeString` instead of `StringBuilder` and be more efficient.

darinadler commented on Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp:
```diff
@@ -519,29 +519,29 @@ bool CacheStorageManager::isActive()
         isFirst = false;
-        builder.append("\"");
+        builder.append("\""_s);
         builder.append(cache->name());
-        builder.append("\"");
+        builder.append("\""_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp:
```diff
@@ -519,29 +519,29 @@ bool CacheStorageManager::isActive()
         isFirst = false;
-        builder.append("\"");
+        builder.append("\""_s);
         builder.append(cache->name());
-        builder.append("\"");
+        builder.append("\""_s);
```
This could use a single call to `append` and be more efficient.

darinadler commented on Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
```diff
@@ -1828,7 +1828,7 @@ void NetworkStorageManager::cacheStorageRepresentation(CompletionHandler<void(St
         auto fetchedTypes = originStorageManager(origin).fetchDataTypesInList(targetTypes, false);
         if (!fetchedTypes.isEmpty()) {
             StringBuilder originBuilder;
-            originBuilder.append("\n{ \"origin\" : { \"topOrigin\" : \"", origin.topOrigin.toString(), "\", \"clientOrigin\": \"", origin.clientOrigin.toString(), "\" }, \"caches\" : ");
+            originBuilder.append("\n{ \"origin\" : { \"topOrigin\" : \""_s, origin.topOrigin.toString(), "\", \"clientOrigin\": \""_s, origin.clientOrigin.toString(), "\" }, \"caches\" : "_s);
             originBuilder.append(originStorageManager(origin).cacheStorageManager(*m_cacheStorageRegistry, origin, m_queue.copyRef()).representationString());
             originBuilder.append('}');
```
This could use `makeString` instead of `StringBuilder` and be more efficient.

cdumez committed:
Drop StringBuilder:append() overload taking in a `const char*`
https://bugs.webkit.org/show_bug.cgi?id=273671

Reviewed by Tim Nguyen and Darin Adler.

Drop StringBuilder:append() overload taking in a `const char*`, in favor of the
ones taking in an ASCIILiteral or a std::span.

* Source/JavaScriptCore/API/tests/PingPongStackOverflowTest.cpp:
(PingPongStackOverflowObject_hasInstance):
* Source/JavaScriptCore/bytecode/Instruction.h:
* Source/JavaScriptCore/bytecode/Opcode.cpp:
(JSC::padOpcodeName):
(JSC::OpcodeStats::~OpcodeStats):
* Source/JavaScriptCore/bytecode/Opcode.h:
* Source/JavaScriptCore/bytecompiler/NodesCodegen.cpp:
(JSC::ArrayPatternNode::toString const):
(JSC::RestParameterNode::toString const):
* Source/JavaScriptCore/heap/GCLogging.cpp:
(JSC::GCLogging::levelAsString):
* Source/JavaScriptCore/heap/GCLogging.h:
* Source/JavaScriptCore/heap/HeapSnapshotBuilder.cpp:
(JSC::HeapSnapshotBuilder::json):
* Source/JavaScriptCore/jsc.cpp:
(JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/runtime/ConsoleClient.cpp:
(JSC::appendMessagePrefix):
(JSC::ConsoleClient::printConsoleMessage):
(JSC::ConsoleClient::printConsoleMessageWithArguments):
* Source/JavaScriptCore/runtime/DateConversion.cpp:
(JSC::formatDateTime):
* Source/JavaScriptCore/runtime/ExceptionHelpers.cpp:
(JSC::notAFunctionSourceAppender):
* Source/JavaScriptCore/runtime/FileBasedFuzzerAgent.cpp:
(JSC::FileBasedFuzzerAgent::getPredictionInternal):
* Source/JavaScriptCore/runtime/FileBasedFuzzerAgentBase.cpp:
(JSC::FileBasedFuzzerAgentBase::createLookupKey):
* Source/JavaScriptCore/runtime/FunctionConstructor.cpp:
(JSC::stringifyFunction):
* Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp:
(JSC::IntlDateTimeFormat::buildSkeleton):
(JSC::IntlDateTimeFormat::createDateIntervalFormatIfNecessary):
* Source/JavaScriptCore/runtime/IntlDurationFormat.cpp:
(JSC::collectElements):
* Source/JavaScriptCore/runtime/IntlNumberFormat.cpp:
(JSC::IntlNumberFormat::initializeNumberFormat):
* Source/JavaScriptCore/runtime/IntlNumberFormatInlines.h:
(JSC::appendNumberFormatDigitOptionsToSkeleton):
* Source/JavaScriptCore/runtime/Options.cpp:
(JSC::asString):
(JSC::Options::executeDumpOptions):
(JSC::Options::dumpAllOptions):
(JSC::Options::dumpAllOptionsInALine):
(JSC::Options::dumpOption):
(JSC::OptionsHelper::Option::dump const):
* Source/JavaScriptCore/runtime/Options.h:
(JSC::Options::dumpAllOptions):
* Source/JavaScriptCore/runtime/PredictionFileCreatingFuzzerAgent.cpp:
(JSC::PredictionFileCreatingFuzzerAgent::getPredictionInternal):
* Source/JavaScriptCore/runtime/RegExp.cpp:
(JSC::appendLineTerminatorEscape<UChar>):
* Source/JavaScriptCore/runtime/TypeProfiler.cpp:
(JSC::TypeProfiler::typeInformationForExpressionAtOffset):
* Source/JavaScriptCore/runtime/TypeSet.cpp:
(JSC::TypeSet::dumpTypes const):
(JSC::TypeSet::toJSONString const):
(JSC::StructureShape::propertyHash):
(JSC::StructureShape::stringRepresentation):
(JSC::StructureShape::toJSONString const):
* Source/JavaScriptCore/tools/FunctionOverrides.cpp:
(JSC::parseClause):
* Source/WTF/wtf/JSONValues.cpp:
(WTF::JSONImpl::Value::writeJSONImpl const):
* Source/WTF/wtf/text/StringBuilder.h:
(WTF::StringBuilder::append):
* Source/WTF/wtf/text/TextStream.cpp:
(WTF::TextStream::operator<<):
* Source/WebCore/Modules/airplay/WebMediaSessionManager.cpp:
(WebCore::mediaProducerStateString):
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp:
(WebCore::MediaKeySession::updateKeyStatuses):
* Source/WebCore/Modules/encryptedmedia/NavigatorEME.cpp:
(WTF::LogArgument<Vector<T>>::toString):
* Source/WebCore/Modules/fetch/FetchHeaders.cpp:
(WebCore::FetchHeaders::get const):
* Source/WebCore/Modules/geolocation/Geolocation.cpp:
(WebCore::logError):
* Source/WebCore/Modules/indexeddb/IDBKeyData.cpp:
(WebCore::IDBKeyData::loggingString const):
* Source/WebCore/Modules/indexeddb/IDBKeyPath.cpp:
(WebCore::loggingString):
* Source/WebCore/Modules/indexeddb/server/IndexValueStore.cpp:
(WebCore::IDBServer::IndexValueStore::loggingString const):
* Source/WebCore/Modules/indexeddb/shared/IDBDatabaseInfo.cpp:
(WebCore::IDBDatabaseInfo::loggingString const):
* Source/WebCore/Modules/indexeddb/shared/IDBObjectStoreInfo.cpp:
(WebCore::IDBObjectStoreInfo::loggingString const):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCUtils.cpp:
(WebCore::toRTCCodecParameters):
* Source/WebCore/Modules/websockets/WebSocket.cpp:
(WebCore::encodeProtocolString):
* Source/WebCore/Modules/websockets/WebSocketExtensionDispatcher.cpp:
(WebCore::WebSocketExtensionDispatcher::createHeaderValue const):
(WebCore::WebSocketExtensionDispatcher::appendAcceptedExtension):
* Source/WebCore/PAL/pal/avfoundation/OutputContext.mm:
(PAL::OutputContext::deviceName):
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::callInWorld):
* Source/WebCore/contentextensions/CombinedURLFilters.cpp:
(WebCore::ContentExtensions::prefixTreeVertexToString):
(WebCore::ContentExtensions::recursivePrint):
* Source/WebCore/contentextensions/Term.h:
(WebCore::ContentExtensions::Term::toString const):
* Source/WebCore/css/CSSBasicShapes.cpp:
(WebCore::buildEllipseString):
(WebCore::buildInsetString):
* Source/WebCore/css/CSSContainerRule.cpp:
(WebCore::CSSContainerRule::cssText const):
* Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
(WebCore::CSSCounterStyleDescriptors::rangesCSSText const):
(WebCore::CSSCounterStyleDescriptors::symbolsCSSText const):
(WebCore::CSSCounterStyleDescriptors::additiveSymbolsCSSText const):
* Source/WebCore/css/CSSFontFaceSrcValue.cpp:
(WebCore::CSSFontFaceSrcResourceValue::customCSSText const):
* Source/WebCore/css/CSSFontFeatureValuesRule.cpp:
(WebCore::CSSFontFeatureValuesRule::cssText const):
* Source/WebCore/css/CSSFontFeatureValuesRule.h:
* Source/WebCore/css/CSSFontPaletteValuesRule.cpp:
(WebCore::CSSFontPaletteValuesRule::overrideColors const):
(WebCore::CSSFontPaletteValuesRule::cssText const):
* Source/WebCore/css/CSSGradientValue.cpp:
(WebCore::appendGradientStops):
(WebCore::CSSLinearGradientValue::customCSSText const):
(WebCore::CSSPrefixedLinearGradientValue::customCSSText const):
(WebCore::CSSDeprecatedLinearGradientValue::customCSSText const):
(WebCore::CSSRadialGradientValue::customCSSText const):
(WebCore::CSSPrefixedRadialGradientValue::customCSSText const):
(WebCore::CSSDeprecatedRadialGradientValue::customCSSText const):
(WebCore::CSSConicGradientValue::customCSSText const):
* Source/WebCore/css/CSSGroupingRule.cpp:
(WebCore::CSSGroupingRule::appendCSSTextForItemsInternal const):
(WebCore::CSSGroupingRule::cssTextForRules const):
(WebCore::CSSGroupingRule::cssTextForRulesWithReplacementURLs const):
* Source/WebCore/css/CSSImportRule.cpp:
(WebCore::CSSImportRule::cssTextInternal const):
* Source/WebCore/css/CSSLayerBlockRule.cpp:
(WebCore::CSSLayerBlockRule::cssText const):
* Source/WebCore/css/CSSLayerStatementRule.cpp:
(WebCore::CSSLayerStatementRule::cssText const):
* Source/WebCore/css/CSSLineBoxContainValue.cpp:
(WebCore::CSSLineBoxContainValue::customCSSText const):
* Source/WebCore/css/CSSMarkup.cpp:
(WebCore::serializeURL):
* Source/WebCore/css/CSSMediaRule.cpp:
(WebCore::CSSMediaRule::cssText const):
(WebCore::CSSMediaRule::cssTextWithReplacementURLs const):
* Source/WebCore/css/CSSNamespaceRule.cpp:
(WebCore::CSSNamespaceRule::cssText const):
* Source/WebCore/css/CSSPropertyRule.cpp:
(WebCore::CSSPropertyRule::cssText const):
* Source/WebCore/css/CSSScopeRule.cpp:
(WebCore::CSSScopeRule::cssText const):
* Source/WebCore/css/CSSSelectorList.cpp:
(WebCore::CSSSelectorList::buildSelectorsText const):
* Source/WebCore/css/CSSStyleRule.cpp:
(WebCore::CSSStyleRule::cssTextForRules const):
(WebCore::CSSStyleRule::cssTextForRulesWithReplacementURLs const):
(WebCore::CSSStyleRule::cssTextInternal const):
* Source/WebCore/css/CSSStyleSheet.cpp:
(WebCore::CSSStyleSheet::cssTextWithReplacementURLs):
* Source/WebCore/css/CSSSupportsRule.cpp:
(WebCore::CSSSupportsRule::cssText const):
(WebCore::CSSSupportsRule::cssTextWithReplacementURLs const):
* Source/WebCore/css/CSSTimingFunctionValue.cpp:
(WebCore::CSSLinearTimingFunctionValue::customCSSText const):
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::ShorthandSerializer::serializeBorderRadius const):
(WebCore::ShorthandSerializer::serializeGridTemplate const):
* Source/WebCore/css/calc/CSSCalcOperationNode.cpp:
(WebCore::CSSCalcOperationNode::buildCSSText):
* Source/WebCore/css/parser/CSSParserToken.cpp:
(WebCore::CSSParserToken::serialize const):
* Source/WebCore/css/query/GenericMediaQuerySerialization.cpp:
(WebCore::MQ::serialize):
* Source/WebCore/css/query/MediaQueryParser.cpp:
(WebCore::MQ::serialize):
* Source/WebCore/css/typedom/CSSOMVariableReferenceValue.cpp:
(WebCore::CSSOMVariableReferenceValue::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathInvert.cpp:
(WebCore::CSSMathInvert::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathMax.cpp:
(WebCore::CSSMathMax::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathMin.cpp:
(WebCore::CSSMathMin::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathNegate.cpp:
(WebCore::CSSMathNegate::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathProduct.cpp:
(WebCore::CSSMathProduct::serialize const):
* Source/WebCore/css/typedom/numeric/CSSMathSum.cpp:
(WebCore::CSSMathSum::serialize const):
* Source/WebCore/css/typedom/transform/CSSMatrixComponent.cpp:
(WebCore::CSSMatrixComponent::serialize const):
* Source/WebCore/css/typedom/transform/CSSPerspective.cpp:
(WebCore::CSSPerspective::serialize const):
* Source/WebCore/css/typedom/transform/CSSRotate.cpp:
(WebCore::CSSRotate::serialize const):
* Source/WebCore/css/typedom/transform/CSSScale.cpp:
(WebCore::CSSScale::serialize const):
* Source/WebCore/css/typedom/transform/CSSSkewX.cpp:
(WebCore::CSSSkewX::serialize const):
* Source/WebCore/css/typedom/transform/CSSSkewY.cpp:
(WebCore::CSSSkewY::serialize const):
* Source/WebCore/css/typedom/transform/CSSTranslate.cpp:
(WebCore::CSSTranslate::serialize const):
* Source/WebCore/dom/Element.cpp:
(WebCore::appendAttributes):
* Source/WebCore/dom/FragmentDirectiveGenerator.cpp:
(WebCore::FragmentDirectiveGenerator::generateFragmentDirective):
* Source/WebCore/dom/SecurityContext.cpp:
(WebCore::SecurityContext::parseSandboxPolicy):
* Source/WebCore/dom/Text.cpp:
(WebCore::appendTextRepresentation):
* Source/WebCore/editing/FontShadow.cpp:
(WebCore::serializationForCSS):
* Source/WebCore/editing/markup.cpp:
(WebCore::StyledMarkupAccumulator::appendStyleNodeOpenTag):
(WebCore::StyledMarkupAccumulator::appendStartTag):
(WebCore::StyledMarkupAccumulator::appendEndTag):
(WebCore::serializePreservingVisualAppearanceInternal):
(WebCore::urlToMarkup):
* Source/WebCore/html/FormController.cpp:
(WebCore::formSignature):
* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::HTMLImageElement::completeURLsInAttributeValue const):
* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::restrictionNames):
* Source/WebCore/html/parser/HTMLSrcsetParser.cpp:
(WebCore::replaceURLsInSrcsetAttribute):
* Source/WebCore/inspector/InspectorFrontendAPIDispatcher.cpp:
(WebCore::expressionForEvaluatingCommand):
* Source/WebCore/inspector/InspectorOverlay.cpp:
(WebCore::InspectorOverlay::buildFlexOverlay):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::InspectorStyleSheet::setRuleStyleText):
* Source/WebCore/loader/ResourceLoadStatistics.cpp:
(WebCore::appendBoolean):
(WebCore::appendHashSet):
(WebCore::appendNavigatorAPIOptionSet):
(WebCore::appendScreenAPIOptionSet):
(WebCore::ResourceLoadStatistics::toString const):
* Source/WebCore/loader/archive/mhtml/MHTMLArchive.cpp:
(WebCore::generateRandomBoundary):
(WebCore::MHTMLArchive::generateMHTMLData):
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp:
(WebCore::CaptionUserPreferencesMediaAF::captionsDefaultFontCSS const):
* Source/WebCore/page/PageSerializer.cpp:
(WebCore::PageSerializer::serializeCSSStyleSheet):
* Source/WebCore/platform/Decimal.cpp:
(WebCore::Decimal::toString const):
* Source/WebCore/platform/graphics/AV1Utilities.cpp:
(WebCore::createAV1CodecParametersString):
* Source/WebCore/platform/graphics/CodecUtilities.cpp:
(WebCore::humanReadableStringFromCodecString):
* Source/WebCore/platform/graphics/ColorInterpolationMethod.cpp:
(WebCore::serializationForCSS):
* Source/WebCore/platform/graphics/HEVCUtilities.cpp:
(WebCore::createDoViCodecParametersString):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::SeekTarget::toString const):
* Source/WebCore/platform/graphics/VP9Utilities.cpp:
(WebCore::createVPCodecParametersString):
* Source/WebCore/platform/graphics/avfoundation/InbandTextTrackPrivateAVF.cpp:
(WebCore::InbandTextTrackPrivateAVF::processCueAttributes):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMInstanceFairPlayStreamingAVFObjC.mm:
(WTF::LogArgument<WebCore::CDMInstanceFairPlayStreamingAVFObjC::Keys>::toString):
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateMock.cpp:
(WebCore::MediaRecorderPrivateMock::videoFrameAvailable):
(WebCore::MediaRecorderPrivateMock::audioSamplesAvailable):
(WebCore::MediaRecorderPrivateMock::generateMockCounterString):
* Source/WebCore/platform/mediastream/RealtimeMediaSourceSettings.cpp:
(WebCore::RealtimeMediaSourceSettings::convertFlagsToString):
* Source/WebCore/platform/text/DateTimeFormat.cpp:
(WebCore::DateTimeFormat::quoteAndAppendLiteral):
* Source/WebCore/platform/text/win/LocaleWin.cpp:
(WebCore::LocaleWin::shortTimeFormat):
* Source/WebCore/rendering/RenderTreeAsText.cpp:
(WebCore::quoteAndEscapeNonPrintables):
(WebCore::nodePosition):
* Source/WebCore/svg/SVGPathStringBuilder.cpp:
(WebCore::SVGPathStringBuilder::moveTo):
(WebCore::SVGPathStringBuilder::lineTo):
(WebCore::SVGPathStringBuilder::lineToHorizontal):
(WebCore::SVGPathStringBuilder::lineToVertical):
(WebCore::SVGPathStringBuilder::curveToCubic):
(WebCore::SVGPathStringBuilder::curveToCubicSmooth):
(WebCore::SVGPathStringBuilder::curveToQuadratic):
(WebCore::SVGPathStringBuilder::curveToQuadraticSmooth):
(WebCore::SVGPathStringBuilder::arcTo):
(WebCore::SVGPathStringBuilder::closePath):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::imageLastDecodingOptions):
(WebCore::Internals::dumpMarkerRects):
(WebCore::Internals::mediaSessionRestrictions const):
(WebCore::Internals::pageMediaState):
(WebCore::appendOffsets):
(WebCore::Internals::scrollSnapOffsets):
(WebCore::Internals::dumpStyleResolvers):
* Source/WebDriver/SessionHost.cpp:
(WebDriver::SessionHost::sendCommandToBackend):
* Source/WebDriver/socket/HTTPServerSocket.cpp:
(WebDriver::HTTPRequestHandler::packHTTPMessage const):
* Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp:
(WGSL::Metal::FunctionDefinitionWriter::emitNecessaryHelpers):
(WGSL::Metal::FunctionDefinitionWriter::visit):
(WGSL::Metal::FunctionDefinitionWriter::generatePackingHelpers):
(WGSL::Metal::FunctionDefinitionWriter::emitPackedVector):
(WGSL::Metal::FunctionDefinitionWriter::serializeVariable):
(WGSL::Metal::FunctionDefinitionWriter::visitArgumentBufferParameter):
(WGSL::Metal::visitArguments):
(WGSL::Metal::emitTextureDimensions):
(WGSL::Metal::emitTextureGather):
(WGSL::Metal::emitTextureGatherCompare):
(WGSL::Metal::emitTextureLoad):
(WGSL::Metal::emitTextureSample):
(WGSL::Metal::emitTextureSampleCompare):
(WGSL::Metal::emitTextureSampleGrad):
(WGSL::Metal::emitTextureSampleLevel):
(WGSL::Metal::emitTextureSampleBaseClampToEdge):
(WGSL::Metal::emitTextureSampleBias):
(WGSL::Metal::emitTextureNumLayers):
(WGSL::Metal::emitTextureNumLevels):
(WGSL::Metal::emitTextureNumSamples):
(WGSL::Metal::emitTextureStore):
(WGSL::Metal::emitStorageBarrier):
(WGSL::Metal::emitTextureBarrier):
(WGSL::Metal::emitWorkgroupBarrier):
(WGSL::Metal::emitWorkgroupUniformLoad):
(WGSL::Metal::atomicFunction):
(WGSL::Metal::emitAtomicLoad):
(WGSL::Metal::emitAtomicStore):
(WGSL::Metal::emitAtomicAdd):
(WGSL::Metal::emitAtomicSub):
(WGSL::Metal::emitAtomicMax):
(WGSL::Metal::emitAtomicMin):
(WGSL::Metal::emitAtomicAnd):
(WGSL::Metal::emitAtomicOr):
(WGSL::Metal::emitAtomicXor):
(WGSL::Metal::emitAtomicExchange):
(WGSL::Metal::emitDistance):
(WGSL::Metal::emitLength):
(WGSL::Metal::emitDegrees):
(WGSL::Metal::emitDynamicOffset):
(WGSL::Metal::emitBitcast):
(WGSL::Metal::emitPack2x16Float):
(WGSL::Metal::emitUnpack2x16Float):
(WGSL::Metal::emitPack4xI8):
(WGSL::Metal::emitPack4xI8Clamp):
(WGSL::Metal::emitUnpack4xI8):
(WGSL::Metal::emitPack4xU8):
(WGSL::Metal::emitPack4xU8Clamp):
(WGSL::Metal::emitQuantizeToF16):
(WGSL::Metal::emitRadians):
(WGSL::Metal::emitUnpack4xU8):
(WGSL::Metal::FunctionDefinitionWriter::serializeBinaryExpression):
(WGSL::Metal::FunctionDefinitionWriter::serializeConstant):
* Source/WebGPU/WGSL/Parser.cpp:
* Source/WebKit/NetworkProcess/Classifier/ITPThirdPartyData.cpp:
(WebKit::ITPThirdPartyData::toString const):
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp:
(WebKit::ResourceLoadStatisticsStore::dumpResourceLoadStatistics):
(WebKit::appendBoolean):
(WebKit::appendNextEntry):
(WebKit::ResourceLoadStatisticsStore::appendSubStatisticList const):
(WebKit::ResourceLoadStatisticsStore::resourceToString const):
* Source/WebKit/NetworkProcess/PrivateClickMeasurement/PrivateClickMeasurementDatabase.cpp:
(WebKit::PCM::Database::privateClickMeasurementToStringForTesting const):
(WebKit::PCM::Database::attributionToStringForTesting const):
* Source/WebKit/NetworkProcess/soup/NetworkProcessSoup.cpp:
(WebKit::buildAcceptLanguages):
* Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp:
(WebKit::CacheStorageManager::representationString):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::NetworkStorageManager::cacheStorageRepresentation):
* Source/WebKit/Shared/Gamepad/GamepadData.cpp:
(WebKit::GamepadData::loggingString const):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewRunJavaScriptWithParams):
* Source/WebKit/UIProcess/API/gtk/DropTargetGtk4.cpp:
(WebKit::DropTarget::accept):
* Source/WebKit/UIProcess/Inspector/socket/RemoteInspectorProtocolHandler.cpp:
(WebKit::RemoteInspectorProtocolHandler::platformStartTask):
* Source/WebKit/UIProcess/Launcher/win/ProcessLauncherWin.cpp:
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/UIProcess/WebBackForwardList.cpp:
(WebKit::WebBackForwardList::loggingString):
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::dispatchDidReachLayoutMilestone):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::dumpHistoryItem):
* Tools/TestWebKitAPI/Tests/WTF/CrossThreadTask.cpp:
(TestWebKitAPI::LifetimeLogger::fullName):
* Tools/TestWebKitAPI/Tests/WTF/StringBuilder.cpp:
(TestWebKitAPI::TEST(StringBuilderTest, Append)):
(TestWebKitAPI::TEST(StringBuilderTest, VariadicAppend)):
(TestWebKitAPI::TEST(StringBuilderTest, ToString)):
(TestWebKitAPI::TEST(StringBuilderTest, ToStringPreserveCapacity)):
(TestWebKitAPI::TEST(StringBuilderTest, Clear)):
(TestWebKitAPI::TEST(StringBuilderTest, Array)):
(TestWebKitAPI::TEST(StringBuilderTest, Resize)):
(TestWebKitAPI::TEST(StringBuilderTest, ToAtomString)):
(TestWebKitAPI::TEST(StringBuilderTest, ToAtomStringOnEmpty)):
* Tools/TestWebKitAPI/Tests/WebCore/TextCodec.cpp:
(TestWebKitAPI::testDecode):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestAutomationSession.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebContext.cpp:
(generateHTMLContent):
* Tools/TestWebKitAPI/cocoa/HTTPServer.mm:
(TestWebKitAPI::HTTPResponse::serialize const):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::dumpResourceURL):
(WTR::dumpErrorDescriptionSuitableForTestResult):
(WTR::dumpFrameScrollPosition):
(WTR::InjectedBundlePage::didReceiveTitleForFrame):
(WTR::InjectedBundlePage::willSendRequestForFrame):
(WTR::InjectedBundlePage::didReceiveResponseForResource):
(WTR::InjectedBundlePage::didFinishLoadForResource):
(WTR::InjectedBundlePage::didFailLoadForResource):
* Tools/WebKitTestRunner/InjectedBundle/atspi/AccessibilityUIElementAtspi.cpp:
(WTR::attributesOfElement):
(WTR::AccessibilityUIElement::helpText const):
(WTR::AccessibilityUIElement::attributedStringForRange):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::findAndDumpWorldLeaks):
(WTR::TestController::decideDestinationWithSuggestedFilename):
(WTR::TestController::decidePolicyForNavigationAction):

Canonical link: https://commits.webkit.org/278348@main

darinadler reviewed (approved):
I know this already landed, but I looked at all the `StringBuilder` use and saw all kinds of interesting idioms that could be improved.
